### DISCUSSION TITLE
Feature Request: `read_parquet` add parameter to ignore missing files

### DISCUSSION THREAD

**Discussion by goenning** [duckdb/duckdb#13438] [Ideas]
Why do you want this feature?

I'd like for read_parquet to have something like this and if `202405.parquet` file does not exist, it should just skip it.

```sql
SELECT * FROM read_parquet(['./202404.parquet', './202405.parquet'], ignore_missing = true)
```

This would simplify logic on the client side as we don't have to verify the file existence and offload that to duckdb.

**Comment by mvirts** [comment]
I would add that a general ignore_errors would help me. I have a bunch of parquet files where an unknown number of them are corrupted and I can only find one per scan with duckdb because it errors out.

**Comment by sambcom** [comment]
I like [glob](https://duckdb.org/docs/data/multiple_files/overview.html) to handle `aws s3 ls` without another package. 

```
D select * from glob('*.parquet');
┌─────────────┐
│    file     │
│   varchar   │
├─────────────┤
│ foo.parquet │
└─────────────┘
D select * from glob(['foo.parquet','bar.parquet']);
┌─────────────┐
│    file     │
│   varchar   │
├─────────────┤
│ foo.parquet │
└─────────────┘
```

**Reply by matterda** [reply]
Unfortunately, it only works for local files:
```
D select * from glob(['s3://my-bucket/exists.parquet', 's3://my-bucket/does_not_exist.parquet']);
┌───────────────────────────────────────┐
│                  file                 │
│                 varchar               │
├───────────────────────────────────────┤
│     s3://my-bucket/exists.parquet     │
│ s3://my-bucket/does_not_exist.parquet │
└───────────────────────────────────────┘
```
As far as I am aware, there's not equivalent for S3.

EDIT: `glob()` on S3 works fine if using the glob syntax (`glob('s3://my-bucket/*.parquet')`). With an explicit list of files, however, it just returns the list of files without checking for existence.

**Comment by onnimonni** [comment]
I would also really want to see this for `read_json` and `read_csv` too. Whenever certain resource returns 404, 503, 403 it would be excellent to be able to treat that as an empty file instead of throwing an error.
